From 14c5a7bc18f5b423e4137d8b38a5de06693fda51 Mon Sep 17 00:00:00 2001
From: Benjamin Berg <bberg@redhat.com>
Date: Tue, 1 Sep 2020 14:09:47 +0200
Subject: [PATCH 110/134] Fix path for Lenovo kill-switch heuristic

The path was missing the /sys prefix, rendering the test useless. Add
the appropriate prefix and also a comment explaining why we do this.

(cherry picked from commit 8c283e472fa3947ec66e3560599cdce378ce62a1)
Signed-off-by: Paolo Pisati <paolo.pisati@canonical.com>
(cherry picked from commit eb8d3fb298df3e3a094e3579dd64cd8faf237b56)
Signed-off-by: Paolo Pisati <paolo.pisati@canonical.com>
---
 src/thd_engine.cpp | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/thd_engine.cpp b/src/thd_engine.cpp
index 8d736c8..399791e 100644
--- a/src/thd_engine.cpp
+++ b/src/thd_engine.cpp
@@ -697,7 +697,11 @@ static supported_ids_t id_table[] = {
 };
 
 std::vector<std::string> blocklist_paths {
-	"/devices/platform/thinkpad_acpi/dytc_lapmode",
+	/* Some Lenovo machines have in-firmware thermal management,
+	 * avoid having two entities trying to manage things.
+	 * We may want to change this to dytc_perfmode once that is
+	 * widely available. */
+	"/sys/devices/platform/thinkpad_acpi/dytc_lapmode",
 };
 #endif
 
-- 
2.27.0

